catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -183,9 +203,9 @@ private void writeObject(ObjectOutputStream stream) throws IOException {
          */
         stream.defaultWriteObject();
 
-        stream.writeInt(descriptors.size());
+        stream.writeInt(getAttributeDescriptors().size());
```

Comment:
Could save the result of `getAttributeDescriptors()` to a local variable since you're using it twice.